src/Form/AdobeAnalyticsAdminSettings.php

```diff
@@ -36,11 +37,229 @@ public function buildForm(array $form, FormStateInterface $form_state) {
 
     $config = $this->config('adobe_analytics.settings');
 
+    $form['mode'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Installation mode'),
+      '#default_value' => $config->get('mode') ? $config->get('mode') : 'cdn',
+      '#options' => [
+        'cdn' => $this->t('CDN'),
+        'general' => $this->t('Basic'),
+      ],
+      '#weight' => '-100',
+    ];
+    // CDN form elements.
+    $form['cdn_install_type'] = [
+      '#type' => 'radios',
+      '#title' => t('Installation Type'),
+      '#required' => '1',
+      '#default_value' => $config->get('cdn_install_type') ? $config->get('cdn_install_type') : 'amazon',
+      '#options' => [
+        'amazon' => t('Amazon S3 hosted'),
+        'tag' => t('Tag Manager Tool'),
+      ],
+      '#weight' => '-10',
+      '#states' => [
+        'visible' => [
+          ':input[name="mode"]' => ['value' => 'cdn'],
+        ],
+      ],
+    ];
+
+    $form['amazon_s_code_config'] = [
+      '#weight' => '-9',
+      '#description' => t('Enter the Amazon S3 hosted S code Configuration Path (s_code_config.js) for development and production environments.'),
+      '#type' => 'fieldset',
+      '#title' => $this->t('Omniture S code Configuration Path'),
```

Comment:
The word omniture is no longer used by Adobe. Here `Omniture` should be replaced with `Adobe Analytics`. All occurrences of this should be changed in the codebase.